lib/galaxy/tools/toolbox/base.py

```diff
@@ -745,10 +747,13 @@ def load_tool(self, config_file, guid=None, repository_id=None, use_cached=False
             tool = self.create_tool(config_file=config_file, repository_id=repository_id, guid=guid, **kwds)
             if tool.tool_shed_repository or not guid:
                 self.add_tool_to_cache(tool, config_file)
-        if not tool.id.startswith("__") and self._tool_watcher:
+        if not tool.id.startswith("__"):
             # do not monitor special tools written to tmp directory - no reason
             # to monitor such a large directory.
-            self._tool_watcher.watch_file(config_file, tool.id)
+            if self._tool_watcher:
+                self._tool_watcher.watch_file(config_file, tool.id)
+            if self._tool_config_watcher:
+                [self._tool_config_watcher.watch_file(macro_path) for macro_path in tool._macro_paths]
```

Comment:
Just something I ran in to while looking at another issue today - I was confused as to why a macro was part of the `ToolConfWatcher` (the one that watches tool *panel* configs e.g. `shed_tool_conf.xml`) since I don't have tool watching enabled, and it turns out this was the reason. Presumably this is because the tool watcher only reloads single tools, so this was the quickest way to get macro watching enabled.

Maybe there should be macro watcher, or the tool watcher should register tool IDs with macros so it can reload them instead of using the toolbox reloader? Not a big deal, but a little bit of an unintentional side effect that macros are watched even when tool reloading is off. I guess we could also just case this so the macros aren't added to the watcher if tool watching is not enabled.